var/spack/repos/builtin/packages/espressopp/package.py

```diff
@@ -0,0 +1,80 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class Espressopp(CMakePackage):
+    """ESPResSo++ is an extensible, flexible, fast and parallel simulation
+       software for soft matter research. It is a highly versatile software
+       package for the scientific simulation and analysis of coarse-grained
+       atomistic or bead-spring models as they are used in soft matter research
+    """
+    homepage = "https://espressopp.github.io"
+    url      = "https://github.com/espressopp/espressopp/tarball/v1.9.4"
+
+    version('develop', git='https://github.com/espressopp/espressopp.git', branch='master')
+    version('1.9.4', 'f2a27993a83547ad014335006eea74ea')
+
+    variant('debug', default=False, description='Build debug version')
+    variant('ug', default=False, description='Build user guide')
+    variant('pdf', default=False, description='Build user guide in pdf format')
+    variant('dg', default=False, description='Build developer guide')
+
+    depends_on("cmake@2.8:", type='build')
+    depends_on("mpi", type=('build', 'link', 'run'))
```

Comment:
In general, the default type is fine for most dependencies. So for all of these dependencies that specify `type=('build', 'link', 'run')`, I would remove that.

`py-mpi4py` may be an exception. I usually use `type='nolink'` for Python dependencies.